cli/data.go

```diff
@@ -276,24 +277,33 @@ func (c *AppClient) TabularData(dst string, filter *datapb.Filter) error {
 		if len(mds) == 0 {
 			break
 		}
-		for _, md := range mds {
+		// Map the current response's metadata indexes to those combined across all responses.
+		localToGlobalMDIndex := make(map[int]int)
+		for i, md := range mds {
+			currMDIndex, ok := mdIndexes[md.String()]
+			if ok {
+				localToGlobalMDIndex[i] = currMDIndex
+				continue // Already have this metadata file.
+			}
+			mdIndexes[md.String()] = mdIndex
+			localToGlobalMDIndex[i] = mdIndex
+
 			mdJSONBytes, err := protojson.Marshal(md)
 			if err != nil {
 				return errors.Wrap(err, "error marshaling metadata")
 			}
 			//nolint:gosec
-			mdFile, err := os.Create(filepath.Join(dst, metadataDir, strconv.Itoa(metadataIdx)+".json"))
+			mdFile, err := os.Create(filepath.Join(dst, metadataDir, strconv.Itoa(mdIndex)+".json"))
```

Comment:
Very nit, but I guess we only have to create this file if we haven't seen the metadata before. Could save some time in export.

Reply:
Correct! We `continue` above if we already have this metadata file. Added to the comment above to make this more clear.

Reply:
Oh I see! Thank you :)